src/DataSourceProvider.vue

```diff
@@ -1,19 +1,369 @@
 <template>
-  <div>
-    <h1>Your fliplet-widget-data-source-provider widget will be added here.</h1>
-    <First-Component></First-Component>
-  </div>
+  <section>
+    <div class="data-source-title">
+      <strong>{{ widgetData.dataSourceTitle || 'Select a data source' }}</strong>
+    </div>
+
+    <div v-if="isLoading" class="spinner-holder animated">
+      <div class="spinner-overlay">Loading...</div>
+    </div>
+
+    <div v-else-if="hasError" class="alert alert-danger" role="alert">
+      {{ errorMessage }}
+    </div>
+
+    <div v-else class="main-data-source-provider">
+      <section class="data-source-selector">
+
+        <div v-if="dataSources.length && changeDataSource">
+
+          <Select2
+            :dataSources="dataSources"
+            :selectedDataSource="selectedDataSource"
+            :customOptionView="formatDataSourceOption"
+            :customSearch="customDataSourceSearch"
+            :optionLabelKey="'name'"
+            :optionValueKey="'id'"
+            :selectWithGroups="!!allDataSources.length"
+            @onSelect="setDataSource">
```

Comment:
```suggestion
            @update:selectedDataSource="setDataSource">
```

Or just change the line above to the following and you won't need the event at all.

```js
:selectedDataSource.sync="selectedDataSource"
```

See https://vuejs.org/v2/guide/components-custom-events.html#sync-Modifier